content/blog/2023/2023-09-16-create-openfeign-http-client.md

```diff
@@ -0,0 +1,503 @@
+---
+authors: [sagaofsilence]
+title: "Create an HTTP Client with OpenFeign"
+categories: ["Java"]
+date: 2023-09-16 00:00:00 +0530
+excerpt: "Feign is an HTTP client that allows us to easily make web requests. With its simplified and fluent API, it makes consuming RESTful services a breeze."
+image: images/stock/0134-client_sever.png
+url: create-openfeign-http-client
+---
+
+## Introduction
+Feign is an open-source Java library that simplifies the process of making web requests. It streamlines the implementation of RESTful web services by providing a higher-level abstraction. Feign eliminates the need for boilerplate code, which makes the codebase more readable and maintainable.
+
+## What is Feign?
+Feign is a popular Java HTTP client library that offers several advantages and features, making it a good choice for developers building HTTP-based microservices and applications.
+
+### What is a declarative HTTP client?
+It's a way to make http requests by writing a java interface. Feign generates the actual implementation based on annotations that we provide.
```

Comment:
http -> HTTP ... please check if there are other instances of this

Reply:
OpenFeign is not suitable for non-HTTP protocols.